src/indicators/flooding/floodscan/floodscan.py

```diff
@@ -0,0 +1,150 @@
+
+    def compute_stats_per_area(self, ds, raster_transform, adm_path, data_var="SFED_AREA",percentile_list=[2,4,6,8,10,20]):
+        """
+        Compute statistics on the raster cells per admin area
+        Args:
+            ds: the xarray dataset with values per raster cell
+            raster_transform: the affine transformation of ds
+            adm_path: the path to the admin boundaries shp file
+            data_var: the variable of interest in ds
+            percentile_list: list of thresholds to compute the value x% of the cells is below at
+
+        Returns:
+            df_hist: dataframe with the statistics per admin
+        """
+        # compute statistics on level in adm_path for all dates in ds
+        df_list = []
+        for date in ds.time.values:
+            df = gpd.read_file(adm_path)
+            ds_date = ds.sel(time=date)
+
+            df[["mean_cell", "max_cell", "min_cell"]] = pd.DataFrame(
+                zonal_stats(vectors=df, raster=ds_date[data_var].values, affine=raster_transform, nodata=np.nan))[
+                ["mean", "max", "min"]]
+            #TODO: the percentiles seem to always return 0, even if setting the p to 0.00001. Don't understand why yet..
+            df[[f"percentile_{str(p)}" for p in percentile_list]] = pd.DataFrame(
+                zonal_stats(vectors=df, raster=ds_date[data_var].values, affine=raster_transform, nodata=np.nan,
+                            stats=" ".join([f"percentile_{str(p)}" for p in percentile_list])))[
+                [f"percentile_{str(p)}" for p in percentile_list]]
+
+            df["date"] = pd.to_datetime(date)
+
+            df_list.append(df)
+        df_hist = pd.concat(df_list)
+        df_hist = df_hist.sort_values(by="date")
+        #drop the geometry column, else csv becomes huge
+        df_hist=df_hist.drop("geometry",axis=1)
+
+        return df_hist
+
+    def read_processed_dataset(self,
+        country_name: str,
+        country_iso3: str,
+        adm_level: int,
+    ):
+        filepath = self._get_processed_filepath(
+            country_name=country_name,
+            country_iso3=country_iso3,
+            adm_level=adm_level,
+        )
+        return pd.read_csv(filepath,index_col=False)
+
+    def _write_to_processed_file(
+        self,
+        country_name: str,
+        country_iso3: str,
+        adm_level: int,
+        df: pd.DataFrame,
+    ) -> Path:
+        filepath = self._get_processed_filepath(
+            country_name=country_name,
+            country_iso3=country_iso3,
+            adm_level=adm_level,
+        )
+        Path(filepath.parent).mkdir(parents=True, exist_ok=True)
+        filepath.unlink(missing_ok=True)
+        logger.info(f"Writing to {filepath}")
+        df.to_csv(filepath)
+        return filepath
+
+    def _get_raw_filepath(
+            self,
+    ):
+        #is there an advantage of doing this vs os.path.join?
```

Comment:
@turnerm just explained that this is a relatively newer way of doing things, and it sounds nice. You define the top folder as a `Path()` from `pathlib` , and then you can create paths by using the `/` (os independent). It has as advantage that it makes it more readible. And you can attach methods to it, such as creating the path. So lets discuss that separately, but it might be an idea to make that the go to instead of `os.path.join`